geode-core/src/main/java/org/apache/geode/internal/cache/entries/AbstractRegionEntry.java

```diff
@@ -257,27 +257,17 @@ public void makeTombstone(InternalRegion region, VersionTag version)
       throws RegionClearedException {
     assert region.getVersionVector() != null;
     assert version != null;
-    if (region.getServerProxy() == null && region.getVersionVector()
-        .isTombstoneTooOld(version.getMemberID(), version.getRegionVersion())) {
-      // distributed gc with higher vector version preempts this operation
-      if (!isTombstone()) {
-        basicMakeTombstone(region);
-        region.getCachePerfStats().incTombstoneCount(1);
-      }
-      ((DiskRecoveryStore) region).getRegionMap().removeTombstone(this, version, false, true);
-    } else {
-      if (isTombstone()) {
-        // unschedule the old tombstone
-        region.unscheduleTombstone(this);
-      }
-      setRecentlyUsed(region);
-      boolean newEntry = getValueAsToken() == Token.REMOVED_PHASE1;
-      basicMakeTombstone(region);
-      region.scheduleTombstone(this, version);
-      if (newEntry) {
-        // bug #46631 - entry count is decremented by scheduleTombstone but this is a new entry
-        region.getCachePerfStats().incEntryCount(1);
-      }
+    if (isTombstone()) {
```

Comment:
This is fine but I think it would be slightly better to record that this was a tombstone and then when you schedule it use scheduleTombstone(RegionEntry, versionTag, wasTombstone).  That will leave the tombstone count alone and be slightly faster since you don't have to decrement/increment an atomic boolean and record stats.

Reply:
Good point. The new commit has changes as suggested.